validator/sawtooth_validator/journal/block_validator.py

```diff
@@ -323,8 +321,8 @@ def validate_block(self, blkw, consensus, chain=None):
                 # since changes to the chain-head can change the state of the
                 # blocks in BlockStore we have to revalidate this block.
                 block_store = self._block_cache.block_store
-                if self._chain_head is not None and\
-                        self._chain_head.identifier !=\
+                if chain_head is not None and\
```

Comment:
chain_head is set above, thus will never be None here.

Reply:
This is not actually true. When the chain head is retrieved from the block store, it can still be None if there isn't a genesis block yet.